### PR TITLE
Add fix tests

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes :

Added tests for filters and fixed other tests due to change in ui

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/myData/MyDataPageTest.java [hunk 1]
```diff
@@ -121,15 +121,15 @@ public void checkHeaders() {
...
     Events.click(webDriver, By.cssSelector("[data-testid='menu-item-Docs']"));
     webDriver.switchTo().window(tabs.get(0));
 
-    Events.click(webDriver, By.cssSelector("[data-testid='menu-button'][id='menu-button-Need Help']"));
+    Events.click(webDriver, By.cssSelector("[data-testid='menu-button'][id='menu-button-undefined']"));
     Events.click(webDriver, By.cssSelector("[data-testid='menu-item-API']"));
     webDriver.navigate().back();
 
-    Events.click(webDriver, By.cssSelector("[data-testid='menu-button'][id='menu-button-Need Help']"));
+    Events.click(webDriver, By.cssSelector("[data-testid='menu-button'][id='menu-button-undefined']"));
```
Comment (shahsank3t) [inline openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/myData/MyDataPageTest.java:132]: this `undefined` doesn't seem to be correct.
Please check with @Sachin-chaurasiya if this is correct or needs to change on UI side as well to have proper id

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/common/PaginationAndFilterTest.java [hunk 2]
```diff
@@ -82,6 +82,45 @@ public void checkFlikerInFilter() throws Exception {
...
+      }
+    } catch (TimeoutException exception) {
+      LOG.info("Success");
+    }
+  }
+
+  @RepeatedIfExceptionsTest(repeats = 2)
+  @Order(3)
+  public void dataPresentWithFilter() throws Exception {
+    Thread.sleep(waitTime);
```
Comment (shahsank3t) [inline openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/common/PaginationAndFilterTest.java:107]: do we really need `waitTime` here?

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/myData/MyDataPageTest.java (hunk 1 above)
Comment (parthp2107) [reply openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/myData/MyDataPageTest.java:132]: Changed it back to 'Need Help'. Addressed in the latest commit.

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/common/PaginationAndFilterTest.java (hunk 2 above)
Comment (parthp2107) [reply openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/common/PaginationAndFilterTest.java:107]: Removed. Addressed in the latest commit.

File: openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/myData/MyDataPageTest.java (hunk 1 above)
Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/myData/MyDataPageTest.java:132]: @shahsank3t This is not correct so I have shared the patch with @parthp2107 .

**Review by shahsank3t — APPROVED** [review]

**Pull request merged**